### ISSUE TITLE
Implementation of OIDC Authentication in sample-project-aws-serverless

### ISSUE THREAD

**Issue by kdesao-devops** [bcgov/cloud-pathfinder#2116]
**Describe the Issue**
We want to show deployment architecture examples that use the Github OpenID authentication with AWS.
For that we want to implement sample config in the sample apps.
There will be one issue per app, this ticket is focused on the serverless demo app

**Additional Context**
- [Repository](https://github.com/bcgov/startup-sample-project-aws-serverless)
- Blocked by the implementation of the OIDC module
- This ticket is low priority compared to the ECF Upgrade - if the team runs into trouble, we drop this and run over to help

**Acceptance Criteria**
- work should be done on a branch called aws-oidc-auth
- The terraform state has been configured on S3
- The terraform lock has been configured on DynamoDB
- The sample App is deployed on ECF Forge through OIDC using github action workflows

**Comment by kdesao-devops** [comment]
The Serverless app has been deployed on lz0 using the OIDC Authenctication and the configured s3 and dynamoDB Terraform Backend, Here is the PR:

https://github.com/bcgov/startup-sample-project-aws-serverless-OIDC/pull/2
